src/main/java/ch/njol/skript/util/LocationUtils.java

```diff
@@ -0,0 +1,80 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.util;
+
+import org.bukkit.Location;
+
+/**
+ * A class that contains methods based around
+ * making it easier to deal with {@link Location}
+ * objects.
+ */
+public class LocationUtils {
+
+	/**
+	 * Normalizes the location by ensuring that the pitch is between -90 and 90, the yaw is between -180 and 180,
+	 * and all -0.0 values are converted to 0.0.
+	 *
+	 * @param location The location to normalize
+	 * @return The same location with normalized values.
+	 */
+	public static Location normalize(Location location) {
+		location.setPitch(Location.normalizePitch(location.getPitch()));
+		location.setYaw(Location.normalizeYaw(location.getYaw()));
+
+		if (location.getYaw() == -0.0) location.setYaw(0);
+		if (location.getPitch() == -0.0) location.setPitch(0);
+		if (location.getX() == -0.0) location.setX(0);
+		if (location.getY() == -0.0) location.setY(0);
+		if (location.getZ() == -0.0) location.setZ(0);
```

Comment:
Unsure if it'll matter too much but I know we complain about in syntax classses.
```suggestion
		if (location.getYaw() == -0.0)
			location.setYaw(0);
		if (location.getPitch() == -0.0)
			location.setPitch(0);
		if (location.getX() == -0.0)
			location.setX(0);
		if (location.getY() == -0.0)
			location.setY(0);
		if (location.getZ() == -0.0)
			location.setZ(0);
```